sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1874,36 +1878,62 @@ export class ApplicationService extends RecordDataModelService<Application> {
   /**
    * Get previous application versions for an application.
    * @param applicationId application id.
-   * @param options method options.
-   * - `limit` number of previous application versions to be returned.
-   * - `loadDynamicData` optionally loads the dynamic data.
    * @returns previous application versions.
    */
   async getPreviousApplicationVersions(
     applicationId: number,
-    options?: { loadDynamicData?: boolean; limit?: number },
   ): Promise<Application[]> {
-    const application = await this.repo.findOne({
-      select: { applicationNumber: true, submittedDate: true },
-      where: { id: applicationId },
-    });
     return this.repo.find({
       select: {
         id: true,
         submittedDate: true,
-        data: !!options?.loadDynamicData as unknown,
       },
       where: {
-        submittedDate: LessThan(application.submittedDate),
-        applicationNumber: application.applicationNumber,
+        parentApplication: { id: applicationId },
+        applicationStatus: ApplicationStatus.Overwritten,
       },
       order: {
         submittedDate: "DESC",
       },
-      take: options?.limit,
     });
   }
 
+  /**
+   * Get last application version.
+   * @param precedingApplicationId application id of the previous application.
+   * @returns last application version.
+   */
+  async getLastApplicationVersion(
+    precedingApplicationId: number,
+  ): Promise<Application> {
+    return this.repo.findOne({
+      select: { id: true, data: true as unknown },
+      where: { id: precedingApplicationId },
+    });
+  }
+
+  /**
+   * Gets the latest application by parent application id.
+   * @param parentApplicationId parent application id.
+   * @returns the application id.
+   */
+  async getApplicationIdByParentApplicationId(
+    parentApplicationId: number,
+  ): Promise<number> {
+    const [application] = await this.repo.find({
+      select: { id: true, submittedDate: true },
+      where: {
+        parentApplication: {
+          id: parentApplicationId,
+        },
+        applicationStatus: Not(ApplicationStatus.Overwritten),
+      },
+      order: { submittedDate: "DESC" },
+      take: 1,
+    });
+    return application?.id;
```

Comment:
I would say, please return the application object instead of just returning the id to keep the service more generic. Also rename the method to be `getCurrentApplicationByParentApplicationId` or anything close to the logic.

Reply:
The comment is still valid.